src/p2p/net_peerlist_boost_serialization.h

```diff
@@ -134,10 +134,11 @@ namespace boost
       a & port;
       a & length;
 
-      if (length > net::tor_address::buffer_size())
+      const size_t buffer_size = net::tor_address::buffer_size();
+      if (length > buffer_size)
```

Comment:
How does this fix a crash with GCC 9.1? The value returned by `buffer_size()` is a constant, so storing the return value on the stack locally should not have any change in behavior. I might need to install GCC 9 early on my Gentoo devbox ...

Reply:
gcc 9.1 definitely crashes without this patch.

Reply:
It is a GCC ICE, the code was correct AFAICT.

Reply:
Btw, we reported the bug upstream https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90825